modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Index.cshtml.cs

```diff
@@ -47,7 +50,8 @@ public virtual async Task<ActionResult> OnGetAsync()
 
             try
             {
-                Blog = await _blogAppService.GetByShortNameAsync(BlogShortName);
+                Blog = await GetBlogAsync(_blogAppService, BlogOptions, BlogShortName);
+                BlogShortName = Blog.ShortName;
             }
             catch (EntityNotFoundException)
             { 
```

Comment:
There is a possible redirection loop problem. Consider, if we specify the **SingleBlogMode** as **true** but forget to specify the **BlogName** for it, then here it will always throw *EntityNotFoundException* and therefore it will redirect to the **/Blogs/Index** page. And on that page, it redirects to the [Posts/Index page](https://github.com/abpframework/abp/pull/19418/files#diff-cc4fce5479bec487ebf801ef9314c7fe10e160b603eaf3d6b0de5866992f87e4R26-R30) if the single blog mode is enabled.